### PR TITLE
Add build metadata file

### PR DESCRIPTION
**Pull request by johnoliver**

### CODE REVIEW

**Comment by johnoliver** [comment]
added an example of some of the data

File: pipelines/build/openjdk_build_pipeline.groovy [hunk 1]
```diff
@@ -144,11 +145,56 @@ def sign(config) {
...
+            variant: config.variant,
+            version: config.javaVersion,
+            tag    : config.parameters.TAG
+    ]
+
+    /*
+    example data:
+    {
+        "os": "linux",
+        "arc": "x64",
```
Comment (karianna) [inline pipelines/build/openjdk_build_pipeline.groovy]: `arc` -> `arch`

File: pipelines/build/openjdk_build_pipeline.groovy [hunk 2]
```diff
@@ -144,11 +145,56 @@ def sign(config) {
...
+        "variant": "hotspot",
+        "version": "jdk8u",
+        "tag": "jdk8u202-b08",
+        "adopt_build_number": 2,
```
Comment (karianna) [inline pipelines/build/openjdk_build_pipeline.groovy]: Should we make this part of the version number?

File: pipelines/build/openjdk_build_pipeline.groovy [hunk 3]
```diff
@@ -144,11 +145,56 @@ def sign(config) {
...
+            os     : config.os,
+            arc    : config.arch,
+            variant: config.variant,
+            version: config.javaVersion,
+            tag    : config.parameters.TAG
+    ]
+
+    /*
+    example data:
+    {
```
Comment (karianna) [inline pipelines/build/openjdk_build_pipeline.groovy]: I wonder if this needs to be in a top-level block.  What if for example, we added a `build_configure` section with all of the `bash configure` output or the versions of all of the O/S native libs we build with etc....

File: pipelines/build/openjdk_build_pipeline.groovy (hunk 2 above)
Comment (johnoliver) [reply pipelines/build/openjdk_build_pipeline.groovy]: well, this is a question of how we would bring this into the version that would be compatible with the upstream version number

Comment (karianna) [reply pipelines/build/openjdk_build_pipeline.groovy]: Did you see https://bugs.openjdk.java.net/browse/JDK-8207849 - that might help - although not as of today I guess.

**Comment by karianna** [comment]
No bot - it needs discussion.

**Pull request closed without merging**